pkg/crypto/keys/publickey_test.go

```diff
@@ -94,6 +94,20 @@ func TestDecodeFromString(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestPublicKey_Cmp(t *testing.T) {
+	var pub1 PublicKey
+	require.NoError(t, pub1.DecodeBytes([]byte{0x00}))
+	pk, err := NewPrivateKey()
+	require.NoError(t, err)
+	pub2 := pk.PublicKey()
+	t.Run("equal", func(t *testing.T) {
+		require.Equal(t, false, pub1.Equal(pub2))
+	})
+	t.Run("cmp", func(t *testing.T) {
+		require.Equal(t, -1, pub1.Cmp(pub2))
+	})
```

Comment:
We need three testcases for this PR in fact:
1. Compare infinity with non-nil key.
2. Compare non-nil key with infinity.
3. Compare infinity with infinity.

Name them accordingly.